### PR TITLE
Reject decoding of an INT64 with a value >INT64_MAX

### PR DESCRIPTION
**Pull request by mattcaswell**
##### Checklist

- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

File: test/asn1_encode_test.c [hunk 1]
```diff
@@ -372,7 +372,7 @@ static ASN1_INT64_DATA int64_expected[] = {
     CUSTOM_EXPECTED_SUCCESS(ASN1_LONG_UNDEF, ASN1_LONG_UNDEF), /* t_zero */
     CUSTOM_EXPECTED_SUCCESS(1, 1), /* t_one */
     CUSTOM_EXPECTED_FAILURE,     /* t_9bytes_1 */
-    CUSTOM_EXPECTED_SUCCESS(INT64_MIN, INT64_MIN), /* t_8bytes_1 */
+    CUSTOM_EXPECTED_FAILURE, /* t_8bytes_1 (too large positive) */
```
Comment (levitte) [inline test/asn1_encode_test.c]: Indent that comment like the line before (our source reformatting set the start of comment at column 33)

**Comment by snhenson** [comment]
I think we need some underflow tests for too large a negative value and appropriate tests. Maybe a separate PR?

File: crypto/asn1/x_int64.c [hunk 2]
```diff
@@ -71,6 +71,11 @@ static int uint64_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
         ASN1err(ASN1_F_UINT64_C2I, ASN1_R_ILLEGAL_NEGATIVE_VALUE);
         return 0;
     }
+    if ((it->size & INTxx_FLAG_SIGNED) == INTxx_FLAG_SIGNED
+            && !neg && utmp > INT64_MAX) {
+        ASN1err(ASN1_F_UINT32_C2I, ASN1_R_TOO_LARGE);
+        return 0;
+    }
```
Comment (dot-asm) [inline crypto/asn1/x_int64.c]: No, that's not what problem is.

File: crypto/asn1/x_int64.c [hunk 3]
```diff
@@ -71,6 +71,11 @@ static int uint64_c2i(ASN1_VALUE **pval, const unsigned char *cont, int len,
         ASN1err(ASN1_F_UINT64_C2I, ASN1_R_ILLEGAL_NEGATIVE_VALUE);
         return 0;
     }
+    if ((it->size & INTxx_FLAG_SIGNED) == INTxx_FLAG_SIGNED
+            && !neg && utmp > INT64_MAX) {
+        ASN1err(ASN1_F_UINT32_C2I, ASN1_R_TOO_LARGE);
```
Comment (dot-asm) [inline crypto/asn1/x_int64.c]: ASN1_F_UINT64_C2I, not ASN1_F_UINT32_C2I.

**Comment by mattcaswell** [comment]
> I think we need some underflow tests for too small a negative value and appropriate tests. Maybe a separate PR?

Yes, I think a different PR

File: test/asn1_encode_test.c (hunk 1 above)
Comment (mattcaswell) [inline test/asn1_encode_test.c]: Done

File: crypto/asn1/x_int64.c (hunk 2 above)
Comment (mattcaswell) [inline crypto/asn1/x_int64.c]: This PR only addresses the issue where an INT64 is decoded with a value >INT64_MAX. Other issues are addressed by #3160

File: crypto/asn1/x_int64.c (hunk 3 above)
Comment (mattcaswell) [inline crypto/asn1/x_int64.c]: Fixed.

**Comment by mattcaswell** [comment]
Updated commit pushed addressing all issues.

**Comment by mattcaswell** [comment]
Pushed to master and 1.1.0
